coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -1749,6 +1801,11 @@ func InsertWorkspaceResource(ctx context.Context, db database.Store, jobID uuid.
 			String: protoResource.InstanceType,
 			Valid:  protoResource.InstanceType != "",
 		},
+		ModulePath: sql.NullString{
+			String: protoResource.ModulePath,
+			// empty string is root module
```

Comment:
Do we explicitly skip over the root module when we extract it from `modules.json`? IIRC `tfaddr.NewAddress` does not consider an empty address valid? 

Do we even _care_ about the root module?

Reply:
We do skip the root module [when we extract it](https://github.com/coder/coder/blob/b72b8ef3545e43dffbb8417d8d040818177e19d7/provisioner/terraform/modules.go#L46). We don't care about it, as it's always present.

I didn't check if `tfaddr.NewAddress` considers an empty string a valid address, but that shouldn't matter here. In normal operation [`convertAddressToModulePath`](https://github.com/coder/coder/blob/b72b8ef3545e43dffbb8417d8d040818177e19d7/provisioner/terraform/resources.go#L781) should never receive an empty string as input, as resources' addresses are non-empty. It can produce an empty string as output, and `ModulePath` here is that output.